include/fastdds/dds/domain/DomainParticipant.hpp

```diff
@@ -720,6 +734,72 @@ class DomainParticipant : public Entity
             const std::string& type_name,
             std::function<void(const std::string& name, const fastrtps::types::DynamicType_ptr type)>& callback);
 
+    /**
+     * Register a custom content filter factory, which can be used to create a ContentFilteredTopic.
+     *
+     * DDS specifies a SQL-like content filter for use by content filtered topics.
+     * If this filter does not meet your filtering requirements, you can register a custom filter.
+     *
+     * To use a custom filter, it must be registered in the following places:
+     * - In any application that uses the custom filter to create a ContentFilteredTopic and the corresponding
+     *   DataReader.
+     * - In each application that writes the data to the applications mentioned above.
+     *
+     * For example, suppose Application A on the subscription side creates a Topic named X and a ContentFilteredTopic
+     * named filteredX (and a corresponding DataReader), using a previously registered content filter, myFilter.
+     * With only that, you will have filtering at the subscription side.
+     * If you also want to perform filtering in any application that publishes Topic X, then you also need to register
+     * the same definition of the ContentFilter myFilter in that application.
+     *
+     * Each @c filter_name can only be used to register a content filter once per DomainParticipant.
+     *
+     * @param filter_name     Name of the filter. Cannot be nullptr, must not exceed 255 characters, and must be
+     *                        unique within this DomainParticipant.
+     * @param filter_factory  Factory of content filters to be registered. Cannot be nullptr.
+     *
+     * @return RETCODE_BAD_PARAMETER if any parameter is nullptr, or the filter_name exceeds 255 characters.
+     * @return RERCODE_PRECONDITION_NOT_MET if the filter_name has been already registered.
+     * @return RETCODE_OK if the filter is correctly registered.
+     */
+    RTPS_DllAPI ReturnCode_t register_content_filter(
+            const char* filter_name,
+            const IContentFilterFactory* filter_factory);
+
+    /**
+     * Lookup a custom content filter factory previously registered with register_content_filter.
```

Comment:
I'm not sure about this. If we allow to get it, should we also allow to register/unregister it? This will allow the user to override the default filter factory, extending it. But it could potentially have a big and dangerous impact.

Reply:
It was finally decided that the user cannot fiddle with `FASTDDS_SQLFILTER_NAME`.